mountpoint-s3-client/src/mock_client.rs

```diff
@@ -53,7 +53,12 @@ pub struct MockClientConfig {
 #[derive(Debug)]
 pub struct MockClient {
     config: MockClientConfig,
-    objects: RwLock<BTreeMap<String, Arc<MockObject>>>,
+    objects: Arc<RwLock<BTreeMap<String, Arc<MockObject>>>>,
```

Comment:
Why have we added Arc here? Like where is the shared ownership between threads used?

Reply:
The reference to the objects map is cloned by a `MockPutObjectRequest` so it can insert the object when the put is completed.